dask_mongo/core.py

```diff
@@ -9,19 +10,39 @@
 from dask.bag import Bag
 from dask.base import tokenize
 from dask.graph_manipulation import checkpoint
-
+from contextlib import contextmanager
 from ._version import __version__
+from functools import lru_cache
 
 appname = f"dask-mongo/{__version__}"
 
+_CLIENTS = {}
+
+
+@contextmanager
```

Comment:
What do you think about removing the context manager and the with-statements? They don't mean anything anymore.